app/code/core/Mage/Catalog/Model/Product.php

```diff
@@ -579,25 +579,23 @@ public function getCategory()
     }
 
     /**
-     * Set assigned category IDs array to product
+     * Replace product's category ID(s)
      *
-     * @param array|string $ids
+     * @param array|int|string $ids the ID(s) as int, comma-separated string or array of ints
      * @return $this
      */
     public function setCategoryIds($ids)
     {
-        if (is_string($ids)) {
+        if (is_string($ids) === true) {
             $ids = explode(',', $ids);
-        } elseif (!is_array($ids)) {
+        } elseif (is_int($ids) === true) {
+            $ids = (array) $ids;
+        } elseif (is_array($ids) === false) {
```

Comment:
```suggestion
        } elseif (is_array($ids)) {
```

Reply:
Is there a coding standard for OpenMage that can resolve this?

Reply:
I guess even phpcbf would dectect this (not tested). Please change.